src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/config/ResourceServerConfiguration.kt

```diff
@@ -2,11 +2,47 @@ package uk.gov.justice.digital.hmpps.incidentreporting.config
 
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.invoke
+import org.springframework.security.config.http.SessionCreationPolicy
+import org.springframework.security.config.web.server.invoke
+import org.springframework.security.web.SecurityFilterChain
+import uk.gov.justice.hmpps.kotlin.auth.AuthAwareTokenConverter
 import uk.gov.justice.hmpps.kotlin.auth.dsl.ResourceServerConfigurationCustomizer
 
 @Configuration
 class ResourceServerConfiguration {
 
+  @Bean
+  fun hmppsSecurityFilterChain(
+    http: HttpSecurity,
+    customizer: ResourceServerConfigurationCustomizer,
+  ): SecurityFilterChain = http {
+    sessionManagement { SessionCreationPolicy.STATELESS }
+    headers { frameOptions { sameOrigin = true } }
+    csrf { disable() }
+    authorizeHttpRequests {
+      customizer.authorizeHttpRequestsCustomizer.dsl
+        // override the entire authorizeHttpRequests DSL
+        ?.also { dsl -> dsl.invoke(this) }
+        // apply specific customizations to the default authorizeHttpRequests DSL
+        ?: also {
+          customizer.unauthorizedRequestPathsCustomizer.unauthorizedRequestPaths.forEach { authorize(it, permitAll) }
+          customizer.anyRequestRoleCustomizer.defaultRole
+            ?.also { authorize(anyRequest, hasRole(it)) }
+            ?: also {
+              authorize("/reports/**", hasRole("VIEW_INCIDENT_REPORTS"))
```

Comment:
Hack to make the reports secure with a role

Reply:
is it temporary? might be worth leaving a to-do message in the code because i wouldn't know which part of this bean method would need to remain and what will need to be removed…?

Reply:
how does this interact with the role annotations elsewhere, do these `authorize(…)` statements replace them?

Reply:
No they supliment them - its just that DPR don't protect their endpoints will a role

Reply:
I'll add a comment